packages/shared/src/domain/address/state.ts

```diff
@@ -1,17 +1,15 @@
+import { BadRequestError } from "../../error/bad-request";
+
 export function normalizeStateSafe(state: string): USState | undefined {
-  if (Object.values(states).includes(USState[state as keyof typeof USState])) {
-    return USState[state as keyof typeof USState];
-  } else if (states[state]) {
-    return states[state];
-  } else if (state === "DC") {
-    return USState.DC;
-  }
-  return undefined;
+  const keyFromEntries = Object.entries(states).find(
+    ([key, value]) => key === state || value.toLowerCase() === state.toLowerCase()
```

Comment:
Why do we have this OR? Couldn't we just have the second one (i.e. they will still be equal after going to lowercase).